File: ISA-CMOP/Features/cv_mdl.m
function [mdl_r2, range_coeff] = cv_mdl(objvar, decvar)

	rmimg = find(sum(imag(objvar)~= 0,2));
	objvar(rmimg,:) = [];
	%decvar(rmimg,:) = [];
    
%------------------Linearity------------------
% Fit a linear model using constraint violation
    mdl = fitlm(decvar,objvar);
% R2
    mdl_r2 = mdl.Rsquared.Adjusted;
% Intercept
%intercept = mdl.Coefficients.Estimate(1);
% Difference between variable coefficient. Large values mean that
% there is at least one variable which is carrying most of the
% weight (given linearity)
    range_coeff = max(mdl.Coefficients.Estimate(2:end))-min(mdl.Coefficients.Estimate(2:end));
end

File: ISA-CMOP_Python/features/cv_mdl.py
import numpy as np
from features.feature_helpers import fit_linear_mdl


def cv_mdl(pop):
    """
    Fit a linear model to decision variables-constraint violation, then take the R2 and difference between the max and min of the absolute values of the linear model coefficients.
    """
    var = pop.extract_var()
    cv = pop.extract_cv()
    obj = pop.extract_obj()

    mdl_r2, range_coeff = fit_linear_mdl(var, cv)

    return mdl_r2, range_coeff
